crates/editor/src/editor.rs

```diff
@@ -4218,16 +4220,42 @@ impl Editor {
             }),
             trigger_kind,
         };
+        let snapshot = buffer.read(cx).snapshot();
+        let classifier = snapshot.char_classifier_at(&buffer_position);
+        let first_char = options
+            .trigger
+            .as_ref()
+            .and_then(|trigger| trigger.chars().next());
+        let should_cancel_backup = first_char.is_some_and(|char| !classifier.is_word(char));
         let completions = provider.completions(&buffer, buffer_position, completion_context, cx);
         let sort_completions = provider.sort_completions();
 
         let id = post_inc(&mut self.next_completion_id);
+        if should_cancel_backup {
+            self.backup_completion_task = id;
+        }
+        dbg!(
+            id,
+            &options.trigger,
+            &should_cancel_backup,
+            self.backup_completion_task
+        );
         let task = cx.spawn(|this, mut cx| {
             async move {
                 this.update(&mut cx, |this, _| {
-                    this.completion_tasks.retain(|(task_id, _)| *task_id >= id);
+                    this.completion_tasks.retain(|(task_id, _)| {
+                        if *task_id >= id || *task_id == this.backup_completion_task {
+                            true
+                        } else {
+                            dbg!("dropping", id);
+                            false
+                        }
+                    });
                 })?;
                 let completions = completions.await.log_err();
+                cx.background_executor()
+                    .timer(Duration::from_millis(1000))
```

Comment:
Doesn't this slow down time to completion for local scenarios?

Reply:
this was just to simulate slow completions